src/main/java/com/couchbase/lite/Database.java

```diff
@@ -3140,39 +3149,40 @@ public boolean sequenceHasAttachments(long sequence) {
                     }
                     else {
                         byte[] data = attachments.blobForKey(key);
-
                         if(data != null) {
                             dataBase64 = Base64.encodeBytes(data);  // <-- very expensive
                         }
                         else {
                             Log.w(Database.TAG, "Error loading attachment.  Sequence: %s", sequence);
                         }
-
                     }
+                }
 
+                String encodingStr = null;
+                if(encoding!=AttachmentInternal.AttachmentEncoding.AttachmentEncodingNone){
```

Comment:
I knew now there is only None and GZip. It is a little safer to check against Gzip.